test/required-headings-match-case.json

```diff
@@ -0,0 +1,13 @@
+{
+  "default": true,
+  "MD043": {
+    "headings": [
+      "# Title",
+      "## First Heading",
+      "## Second Heading",
+      "*",
+      "## Third Heading"
+    ],
+    "match_case": true
```

Comment:
Thanks for adding tests! Please add something to prove this rule will pass on mixed case headings by default and fail on similar input when the new parameter is set.

Reply:
I initially thought that [this test](https://github.com/DavidAnson/markdownlint/blob/main/test/markdownlint-test.js#L631-L687) was sufficient for the cases where the option is turned off (or it is simply not specified). 

Can you share an example of how I can achieve what you mean, please? I am not very familiar with the structure of these tests. Are you suggesting to add a new unit test asserting that the rule fails when the input does not match?

Reply:
The file you added has case in the content match the way the rule is configured and match_case is true. Please add another file where the content does not match case and match_case is true. There should be failures reported which you can tag via the "{MD043}" syntax to show they are expected. There should be a third file (it should already exist, but please add it if not) to verify that content which does not match case is allowed when match_case is not set (i.e., to verify that the code before you changed it allows case to match or not match).

Reply:
I added a new file, can you double check the name and content, please? For the third file, I wonder if https://github.com/DavidAnson/markdownlint/blob/main/test/required-headings-all-present.md is covering that already. What do you think?

Reply:
Agreed, that file covers the default behavior of allowing any case.